### PR TITLE
[backport 2.5] Rollup Interceptor sourceIndex NPE fix (#773) (#781)

### PR DESCRIPTION
**Pull request by lezzago**
*Issue #, if available:*
#764 
*Description of changes:*
Backport of #781 
*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by AWSHurneyt — APPROVED** [review]

File: src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt [hunk 1]
```diff
@@ -109,6 +114,26 @@ class RollupInterceptor(
...
+            logger.warn("Cannot resolve rollup sourceIndex [$sourceIndex]")
+            return ""
+        }
+
+        var concreteIndexName: String = ""
+        if (concreteIndexNames.size == 1 && IndexUtils.isConcreteIndex(concreteIndexNames[0], clusterState)) {
+            concreteIndexName = concreteIndexNames[0]
+        } else if (concreteIndexNames.size > 1) {
+            concreteIndexName = IndexUtils.getNewestIndexByCreationDate(concreteIndexNames, clusterState)
+        } else if (IndexUtils.isAlias(sourceIndex, clusterState) || IndexUtils.isDataStream(sourceIndex, clusterState)) {
```
Comment (amsiglan) [inline src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt:130]: Doubt: Is this condition possible? We have checked for empty, size = 1 and size > 1 above.

File: src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt [hunk 2]
```diff
@@ -109,6 +114,26 @@ class RollupInterceptor(
             }
         }
     }
+
+    fun getConcreteSourceIndex(sourceIndex: String, resolver: IndexNameExpressionResolver, clusterState: ClusterState): String {
+        val concreteIndexNames = resolver.concreteIndexNames(clusterState, IndicesOptions.LENIENT_EXPAND_OPEN, sourceIndex)
+        if (concreteIndexNames.isEmpty()) {
```
Comment (amsiglan) [inline src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt:120]: Should we check for null as well?

**Review by amsiglan — APPROVED** [review]

File: src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt (hunk 2 above)
Comment (lezzago) [reply src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt:120]: I dont think so, it should return empty if it couldn't find anything.

File: src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt (hunk 1 above)
Comment (lezzago) [reply src/main/kotlin/org/opensearch/indexmanagement/rollup/interceptor/RollupInterceptor.kt:130]: That should be applying more to wild card inputs vs alias indices.

**Pull request merged**